Tests/Approval.Tests/ApprovedApi/FluentAssertions/net47.verified.txt

```diff
@@ -461,7 +462,7 @@ namespace FluentAssertions.Collections
         public FluentAssertions.AndConstraint<TAssertions> NotBeSubsetOf(System.Collections.Generic.IEnumerable<T> unexpectedSuperset, string because = "", params object[] becauseArgs) { }
         public FluentAssertions.AndConstraint<TAssertions> NotContain(System.Collections.Generic.IEnumerable<T> unexpected, string because = "", params object[] becauseArgs) { }
         public FluentAssertions.AndConstraint<TAssertions> NotContain(System.Linq.Expressions.Expression<System.Func<T, bool>> predicate, string because = "", params object[] becauseArgs) { }
-        public FluentAssertions.AndWhichConstraint<TAssertions, T> NotContain(T unexpected, string because = "", params object[] becauseArgs) { }
+        public FluentAssertions.AndConstraint<TAssertions> NotContain(T unexpected, string because = "", params object[] becauseArgs) { }
```

Comment:
I'm wondering whether this (breaking) change is related to the AssertionScope feature.
This applies to `GenericDictionaryAssertions` too.
In your PR description you wrote: "Replace all usages of AndWhichConstraint with AndWhich". But this not applies here and I did not find an explanation why this is required or useful in this context.

Reply:
I've updated the PR description with how I ultimately implemented it. The main reason of introducing `AssertionChain` was to allow amending the caller identifier when you chain another assertion using `Which`. See also `upgradingtov7.md`